qemu/tests/nvdimm.py

```diff
@@ -0,0 +1,155 @@
+        if not dimms_expect.issubset(dimms_monitor):
+            invisible_dimms = dimms_expect - dimms_monitor
+            raise error.TestFail("%s dimms are invisible in monitor" % invisible_dimms)
+        check_cmd = "test -b %s" % self.params.get("pmem", "/dev/pmem0")
+        if self.session.cmd_status(check_cmd):
+            raise error.TestFail("check nvdimm in guest failed.")
+        return True
+
+    def format_nvdimm(self):
+        """
+        Format nvdimm device in guest
+        """
+        format_cmd = self.params["format_command"]
+        status, output = self.session.cmd_status_output(format_cmd, timeout=20)
+        if status != 0:
+            raise error.TestFail("Format command failed: %s" % output)
+
+    def mount_nvdimm(self, format_device="yes"):
+        """
+        Mount nvdimm device in guest
+
+        :param format_device: A string to specify if format the device or not
+        """
+        if format_device == "yes":
+            self.format_nvdimm()
+        mount_cmd = self.params['mount_command']
+        status, output = self.session.cmd_status_output(mount_cmd, timeout=10)
+        if status != 0:
+            raise error.TestFail("Mount nvdimm device failed: %s" % output)
+
+    def umount_nvdimm(self):
+        """
+        Umount nvdimm device in guest.
+        """
+        umount_cmd = "umount %s" % self.params['pmem']
+        status, output = self.session.cmd_status_output(umount_cmd, timeout=10)
+        if status != 0:
+            raise error.TestFail("Umount nvdimm device failed: %s" % output)
+
+    def check_calltrace(self):
+        """
+        Check if call trace in guest.
+        """
+        output = self.session.cmd("dmesg -c")
+        if "Call Trace" in output:
+            raise error.TestFail("Guest call trace.")
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Run nvdimm cases:
+    1) Boot guest with nvdimm device backed by a host file
+    2) Login to the guest
+    3) Check nvdimm in monitor and guest
+    4) Format nvdimm device in guest
+    5) Mount nvdimm device in guest
+    6) Create a file in the mount point in guest
+    7) Check the md5 value of the file
+    8) Umount the nvdimm device and check calltrace in guest
+    9) Reboot the guest
+    10) Remount nvdimm device in guest
+    11) Check if the md5 value of the nv_file changes
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    nvdimm_test = NvdimmTest(params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    try:
+        error.context("Login to the guest", logging.info)
+        login_timeout = int(params.get("login_timeout", 360))
+        nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
+        error.context("Verify nvdimm in monitor and guest", logging.info)
+        if not nvdimm_test.verify_nvdimm(vm):
```

Comment:
it looks this function always return true or raise exception, so this check is useless really.